components/kyma-adapter/internal/types/tenantmapping/tenant_mapping.go

```diff
@@ -112,7 +120,16 @@ func (b Body) GetApplicationID() string {
 	return b.AssignedTenant.UclSystemTenantID
 }
 
-// GetApplicationConfiguration returns the Body application configuration
-func (b Body) GetApplicationConfiguration() Configuration {
-	return b.AssignedTenant.Configuration
+// GetApplicationConfiguration returns the Body application configuration as Configuration struct
+func (b Body) GetApplicationConfiguration() (Configuration, error) {
+	if b.AssignedTenant.Configuration != nil && (string(b.AssignedTenant.Configuration) == "" || string(b.AssignedTenant.Configuration) == "{}" || string(b.AssignedTenant.Configuration) == "\"\"" || string(b.AssignedTenant.Configuration) == "null") {
```

Comment:
Then it will continue to the `json.Unmarshal(..)` and it will return an error. I'll fix that and add such a unit test(when the configuration object is missing in the `assignedTenant`). Good catch.